crates/primitives/src/lib.rs

```diff
@@ -113,3 +113,45 @@ pub mod utils {
 pub mod kzg {
     pub use c_kzg::*;
 }
+
+/// Helpers for working with serde
+pub mod serde_helper;
+
+/// Returns the keccak256 hash for the given data.
+#[inline]
+pub fn keccak256(data: impl AsRef<[u8]>) -> H256 {
+    use tiny_keccak::{Hasher, Keccak};
+
+    let mut buf = [0u8; 32];
+    let mut hasher = Keccak::v256();
+    hasher.update(data.as_ref());
+    hasher.finalize(&mut buf);
+    buf.into()
+}
+
+/// Hash a message according to [EIP-191] (version `0x01`).
+///
+/// The final message is a UTF-8 string, encoded as follows:
+/// `"\x19Ethereum Signed Message:\n" + message.length + message`
+///
+/// This message is then hashed using [Keccak-256](keccak256).
+///
+/// [EIP-191]: https://eips.ethereum.org/EIPS/eip-191
+pub fn hash_message<T: AsRef<[u8]>>(message: T) -> H256 {
+    const PREFIX: &str = "\x19Ethereum Signed Message:\n";
+
+    let message = message.as_ref();
+    let len = message.len();
+    let len_string = len.to_string();
+
+    let mut eth_message = Vec::with_capacity(PREFIX.len() + len_string.len() + len);
+    eth_message.extend_from_slice(PREFIX.as_bytes());
+    eth_message.extend_from_slice(len_string.as_bytes());
+    eth_message.extend_from_slice(message);
+
+    H256(*keccak256(&eth_message))
+}
```

Comment:
This feels like common code that we'd want to have as utilities in alloy core?

Reply:
I guess so, in `alloy_primitives::utils`

Reply:
We can merge this and upstream after, if you don't mind opening a PR to [alloy-rs/core](https://github.com/alloy-rs/core) @supernovahs after this PR?